### PR TITLE
New content behaviour fix android (uplift to 1.39.x)

### PR DESCRIPTION
**Pull request by tapanmodh**
Fixes https://github.com/brave/brave-browser/issues/22182
Fixes https://github.com/brave/brave-browser/issues/22183
Fixes https://github.com/brave/brave-browser/issues/22356
Uplift of https://github.com/brave/brave-core/pull/13155

Pre-approval checklist: 
- [ ] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [ ] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [ ] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java [hunk 1]
```diff
@@ -907,6 +895,12 @@ private void initNews() {
                     public void onScrollChanged() {
                         try {
                             int scrollY = mParentScrollView.getScrollY();
+                            SharedPreferencesManager.getInstance().writeInt(
+                                    BRAVE_NESTED_SCROLLVIEW_POSITION
+                                            + BraveActivity.getBraveActivity()
+                                                      .getActivityTab()
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:901]: do we need to check tab on null? as we do check in the code above.

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java [hunk 2]
```diff
@@ -1055,16 +1049,41 @@ public void onClick(View v) {
...
-                                                                + dpToPx(getContext(), 90));
+                                                scrollToY = pxHeight - dpToPx(getContext(), 215);
                                             }
+
+                                            mParentScrollView.smoothScrollTo(0, scrollToY);
+                                            // mRecyclerView.scrollToPosition(0);
+                                            SharedPreferencesManager.getInstance().writeInt(
+                                                    "position_"
+                                                            + BraveActivity.getBraveActivity()
+                                                                      .getActivityTab()
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:1077]: the same about null

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java [hunk 3]
```diff
@@ -1055,16 +1049,41 @@ public void onClick(View v) {
...
+                                                                      .getId(),
+                                                    -1);
+
+                                            SharedPreferencesManager.getInstance().writeInt(
+                                                    BRAVE_RECYCLERVIEW_POSITION
+                                                            + BraveActivity.getBraveActivity()
+                                                                      .getActivityTab()
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:1084]: the same about null

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java [hunk 4]
```diff
@@ -1148,25 +1165,23 @@ public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newStat
...
-                                        position);
-                            }
+                            View firstChild = mRecyclerView.getChildAt(0);
+                            int firstVisiblePosition =
+                                    mRecyclerView.getChildAdapterPosition(firstChild);
+                            int verticalOffset = firstChild.getTop();
+
+                            SharedPreferencesManager.getInstance().writeInt("position_"
+                                            + BraveActivity.getBraveActivity()
+                                                      .getActivityTab()
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:1175]: null check?

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java [hunk 5]
```diff
@@ -1148,25 +1165,23 @@ public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newStat
...
+                                                      .getId(),
+                                    verticalOffset);
+
+                            SharedPreferencesManager.getInstance().writeInt(
+                                    BRAVE_RECYCLERVIEW_POSITION
+                                            + BraveActivity.getBraveActivity()
+                                                      .getActivityTab()
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:1182]: null check?

**Comment by kjozwiak** [comment]
Closing in favour of https://github.com/brave/brave-core/pull/13184.

**Pull request closed without merging**